client/my-sites/sharing/buttons/preview-action.jsx

```diff
@@ -1,52 +1,55 @@
 /**
  * External dependencies
  */
-var React = require( 'react' ),
-	classNames = require( 'classnames' );
+import React, { Component, PropTypes } from 'react';
+import classNames from 'classnames';
+import { omit, noop } from 'lodash';
 
-module.exports = React.createClass( {
-	displayName: 'SharingButtonsPreviewAction',
-
-	propTypes: {
-		active: React.PropTypes.bool,
-		position: React.PropTypes.oneOf( [
+export default class SharingButtonsPreviewAction extends Component {
+	static propTypes = {
+		active: PropTypes.bool,
+		position: PropTypes.oneOf( [
 			'top-left',
 			'top-right',
 			'bottom-left',
 			'bottom-right'
 		] ),
-		icon: React.PropTypes.string,
-		onClick: React.PropTypes.func
-	},
+		icon: PropTypes.string,
+		onClick: PropTypes.func
+	};
+
+	static defaultProps = {
+		active: true,
+		position: 'top-left',
+		onClick: noop
+	};
 
-	getDefaultProps: function() {
-		return {
-			active: true,
-			position: 'top-left',
-			onClick: function() {}
-		};
-	},
+	constructor( props ) {
+		super( props );
+	}
 
-	getIconElement: function() {
-		if ( this.props.icon ) {
-			return <span className={ 'noticon noticon-' + this.props.icon } />;
+	getIconElement() {
+		const { icon } = this.props;
+		if ( icon ) {
+			return <span className={ 'noticon noticon-' + icon } />;
 		}
-	},
+	}
 
-	render: function() {
-		var classes = classNames( 'sharing-buttons-preview-action', {
-			'is-active': this.props.active,
-			'is-top': 0 === this.props.position.indexOf( 'top' ),
-			'is-right': -1 !== this.props.position.indexOf( '-right' ),
-			'is-bottom': 0 === this.props.position.indexOf( 'bottom' ),
-			'is-left': -1 !== this.props.position.indexOf( '-left' )
+	render() {
+		const { active, position, children } = this.props;
+		const classes = classNames( 'sharing-buttons-preview-action', {
+			'is-active': active,
+			'is-top': 0 === position.indexOf( 'top' ),
```

Comment:
[`startsWith`](https://lodash.com/docs/4.16.4#startsWith)  and [`endsWith`](https://lodash.com/docs/4.16.4#endsWith) would be more human-readable alternatives to this.